procd: extract service command from /etc/shint to /sbin

feckert opened the pull request:
Using the service command as a shell alias function has certain disadvantages.
1. if you don't look it up explicitly, you don't know the command, because it is not offered in a tab compilation.
2. The `shinit` file is maintained in the base files. But the command belongs to procd, because it uses its ubus backend. Therefore, in my view, it also belongs to `procd`.

The second commit adds a simplified call for the following ubus command.
`ubus call service list "{ 'verbose': true, 'name': '<service-name>' }"`

It has always bothered me that I had to enter the whole thing if I wanted to have the extended information from the `procd` for this service.

feckert committed:
procd: move service command to procd

The service command belongs to the procd and does not belong in the
shinit. In the course of the move, the script was also checked with
shellcheck and cleaned up.

Signed-off-by: Florian Eckert <fe@dev.tdt.de>

aparcar commented:
I assume the `service` function isn't used anywhere else?

feckert commented:
> I assume the `service` function isn't used anywhere else?
What do you mean?
I use it on the shell to start and stop the services.
I also wanted to use it for the `ubus service list` call as suggest by the second commit.
I often need this output from the ubus during development, so I thought this should also go to the service shell command

bobafetthotmail commented:
> I assume the `service` function isn't used anywhere else?

It was originally introduced as a shortcut for interactive shell use (so you don't have to do **/etc/init.d/something start/stop/whatever**) with this commit https://github.com/openwrt/openwrt/commit/d52676d1ea51bcfc947970085184fe4d18638ccd#diff-180507155f50542221fe10116f13a6c027201b48d18d53c499a975d4e562de55

then it was moved in its current place in a different file https://github.com/openwrt/openwrt/commit/12020f8a95d21547bd7d7fd1fef9cdecf2892803#diff-905995024193d40e44180286139ec5335f3b336b15d7224960f74a822f6be1b4

It's been evolving to become a fully-fledged wrapper using ubus commands.

oldium commented on package/system/procd/files/service:
```diff
@@ -0,0 +1,34 @@
+	local boot status
+
+	if [ -f "/etc/init.d/${service}" ]; then
+		if [ "$cmd" = "list" ]; then
```
This effectively overrides any `list` command from `/etc/init.d/${service}`. I think (I am not an OpenWrt maintainer) it would be better to have this directly in the init script support functions (`/etc/rc.common`), although it would conflict with the `list` command from `lib/functions.sh`, so a different command name would be necessary (like `procd_list`). So that you could call interchangeably `/etc/init.d/my_service procd_list` and `service my_service procd_list`.

feckert replied:
You're right about that. I'll have to look at that again. Thank you for your feedback. 
I will rebase this

feckert committed:
base-files: add wrapper for procd service list command

A service managed by procd does have a json object with usefull information.
This information could by dumped with the following command.

ubus call service list "{ 'verbose':true, 'name': '<service-name>)'". }"

This line is long and complicated to enter. This commit adds a wrapper
call to the procd service section tool to simplify the input and get the
output faster.

We could now enter the command /etc/initd/<service> info to get the info
faster.

Signed-off-by: Florian Eckert <fe@dev.tdt.de>

feckert commented:
@aparcar Can you take a look at this. I find the second commit very useful. 
https://github.com/openwrt/openwrt/pull/4830/commits/c8af0fccfe1927fa6d7225725ddefcab965e4b3d
I have now used the word `info`. I have not seen any service that already uses this.

oldium commented:
I like it - the info command is now supported only on procd-based services.

feckert commented:
@aparcar @oldium @bobafetthotmail ping?

hauke commented:
Thank you for your patches, I applied them to master in b9017384cae7bbc47186fdf35a80207844876a0d and 09c41ea6792a86cfcacb13f6727f5049c3f87cd6.
